src/nsolid/nsolid_api.cc

```diff
@@ -2658,6 +2663,42 @@ uint32_t calculateKernelVersion() {
 }
 #endif
 
+void loadEbpfPrograms() {
+  EbpfLoader loader;
+  Debug(DebugCategory::NSOLID_EBPF, "Attempting to load eBPF programs...\n");
+
+  EbpfLoadStatus status = loader.LoadProgram("hello_world");
+
+  switch (status) {
+    case EbpfLoadStatus::SUCCESS:
+      Debug(DebugCategory::NSOLID_EBPF,
+            "eBPF program 'hello_world' loaded successfully.\n");
+      break;
+    case EbpfLoadStatus::FILE_NOT_FOUND:
+      Debug(DebugCategory::NSOLID_EBPF,
+            "Error - eBPF program 'hello_world' file not found.\n");
+      break;
+    case EbpfLoadStatus::NOT_ALLOWED:
+      Debug(DebugCategory::NSOLID_EBPF,
+            "Error - eBPF program 'hello_world' is not in the allowed list.\n");
+      break;
+    case EbpfLoadStatus::LOAD_ERROR:
+      Debug(DebugCategory::NSOLID_EBPF,
+            "Error - Failed to load eBPF program 'hello_world'.\n");
+      break;
+    case EbpfLoadStatus::NOT_SUPPORTED:
+      Debug(DebugCategory::NSOLID_EBPF,
+            "Failed to load: either eBPF is not supported or missing required "
+            "privileges (e.g., not running with sudo)\n");
+      break;
+    default:
+      Debug(
+          DebugCategory::NSOLID_EBPF,
+          "Error - Unknown status while loading eBPF program 'hello_world'.\n");
+      break;
+  }
+}
+
```

Comment:
This code seems misplaced here. It should be part of `EnvList` interface, imho

Reply:
I was assuming we'll need to load ebpf programs during runtime (not only during initialization). That's why I added to nsolid_api, to have a JS API for it.